scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedJvmTaskExecutionCommandCreator.java

```diff
@@ -0,0 +1,124 @@
+ * Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduler.task.executors.forked.env;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.ow2.proactive.resourcemanager.utils.OneJar;
+import org.ow2.proactive.scheduler.common.task.ForkEnvironment;
+import org.ow2.proactive.scheduler.common.util.VariableSubstitutor;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.scheduler.task.context.TaskContext;
+import org.ow2.proactive.scheduler.task.context.TaskContextVariableExtractor;
+import org.ow2.proactive.scheduler.task.executors.forked.env.command.JavaPrefixCommandExtractor;
+import org.ow2.proactive.scripting.ForkEnvironmentScriptResult;
+import org.ow2.proactive.scripting.ScriptResult;
+import com.google.common.base.Strings;
+
+public class ForkedJvmTaskExecutionCommandCreator implements Serializable {
+    private final static String javaHomePostfixJavaExecutable = File.separatorChar + "bin" + File.separatorChar + "java";
+    private final TaskContextVariableExtractor taskContextVariableExtractor = new TaskContextVariableExtractor();
+    private final JavaPrefixCommandExtractor javaPrefixCommandExtractor = new JavaPrefixCommandExtractor();
+
+    /**
+     * Creates a command to start a task inside a java virtual machine.
+     *
+     * @param taskContext                   TaskContext object describing the task.
+     * @param forkEnvironmentScriptResult   Result from a running fork environment script. If it is
+     *                                      of instance {@link ForkEnvironmentScriptResult}, the script return
+     *                                      variables will be used for the construction of the command.
+     * @param serializedContextAbsolutePath The serialized TaskContext object which will be read by the virtual
+     *                                      machine to run the task.
+     * @return A List, empty if the TaskContext is null, otherwise filled with a command.
+     * @throws Exception If the {@link TaskContextVariableExtractor} could not extract all variables from the
+     *                   TaskContext.
+     */
+    public List<String> createForkedJvmTaskExecutionCommand(TaskContext taskContext,
+            ScriptResult forkEnvironmentScriptResult, String serializedContextAbsolutePath) throws Exception {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)